caiman/gui/gui_pyqtgraph_layout.py

```diff
@@ -235,19 +348,146 @@ def mouseClickEvent(event):
                  {'name': 'REMOVE SINGLE', 'type': 'action'},
                  {'name': 'SAVE OBJECT', 'type': 'action'}
                  ]
-    
+
 
 pars_action = Parameter.create(name='params_action', type='group', children=params_action) 
    
 t_action.setParameters(pars_action, showTop=False)
 t_action.setWindowTitle('Parameter Action')
 
+
+def reset_button():
+    global mode
+    mode = "reset"
+    p2.setTitle("mode: %s" % (mode))
+    #clear the upper right image
+    zeros = np.asarray([ [0] * 80 for _ in range(60)])
+    img2.setImage(make_color_img(zeros), autoLevels=False)
+    draw_contours()
+    
+
+pars.param('RESET').sigActivated.connect(reset_button)
+
+
+def show_background_button():
+    global bg_vline, min_background, max_background, background_num
+    global mode, background_first_frame_scaled
+    #clear thhe upper right image
+    zeros = np.asarray([ [0] * 80 for _ in range(60)])
+    img2.setImage(make_color_img(zeros), autoLevels=False)
+    
+    background_num = (background_num + 1) % estimates.f.shape[0]
+    mode = "background"
+    p2.setTitle("mode: %s %d" % (mode,background_num))
+
+    # display the first frame of the background
+    background_first_frame = estimates.b[:,background_num].reshape(estimates.dims, order='F')
+    min_background_first_frame = np.min(background_first_frame)
+    max_background_first_frame = np.max(background_first_frame)
+    background_first_frame_scaled = make_color_img(background_first_frame, min_max=(min_background_first_frame, max_background_first_frame))
+    img.setImage(background_first_frame_scaled,autoLevels=False)
+    
+    # draw the trace and the infinite line
+    trace_background = estimates.f[background_num]
+    p2.plot(trace_background, clear=True)
+    bg_vline = pg.InfiniteLine(angle = 90, movable = True)
+    p2.addItem(bg_vline, ignoreBounds=True)
+    bg_vline.setValue(0)
+    bg_vline.sigPositionChanged.connect(show_background_update)
+    
+
+def show_background_update():
+    global bg_index, min_background, max_background, background_scaled
+    bg_index = int(bg_vline.value())
+    if bg_index > 0 and bg_index < 2001:
```

Comment:
Instead of 2001 you should have something like `estimates.f.shape[-1]`.
Also is the indexing from 1 to T or is it pythonic from 0 to T-1?

Reply:
I agree. I believe it should be pythonic from 0 to T-1. Thx!